tree_hash/src/impls.rs

```diff
@@ -186,6 +187,24 @@ impl TreeHash for H256 {
     }
 }
 
+impl<T: TreeHash> TreeHash for Arc<T> {
+    fn tree_hash_type() -> TreeHashType {
+        T::tree_hash_type()
+    }
+
+    fn tree_hash_packed_encoding(&self) -> PackedEncoding {
+        self.as_ref().tree_hash_packed_encoding()
+    }
+
+    fn tree_hash_packing_factor() -> usize {
+        T::tree_hash_packing_factor()
+    }
+
+    fn tree_hash_root(&self) -> Hash256 {
+        self.as_ref().tree_hash_root()
```

Comment:
Do you mind adding a unit test for this, just because these sorts of implementations can be accidentally recursive quite easily. The compiler should warn us, but better to be sure with a test

Reply:
Done, thanks!